tests/personio/test_personio_source.py

```diff
@@ -0,0 +1,35 @@
+import dlt
+import pytest
+
+from sources.personio import personio_source
+from tests.utils import ALL_DESTINATIONS, assert_load_info, load_table_counts
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_all_resources(destination_name: str) -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="personio",
+        destination=destination_name,
+        dataset_name="shopify_data",
+        full_refresh=True,
+    )
+    # Set per page limit to ensure we use pagination
+    tables = ["employees", "absences"]
+    load_info = pipeline.run(personio_source(items_per_page=10).with_resources(*tables))
+    print(load_info)
+    assert_load_info(load_info)
+    table_names = [t["name"] for t in pipeline.default_schema.data_tables()]
+    table_counts = load_table_counts(pipeline, *table_names)
+
+    # for now only check main tables
+    assert set(table_counts.keys()) > set(tables)
+    assert table_counts["employees"] >= 31
+    assert table_counts["absences"] >= 6
+
+    # load again to check there are no dupicates
+    load_info = pipeline.run(personio_source(items_per_page=10).with_resources(*tables))
+    table_names = [t["name"] for t in pipeline.default_schema.data_tables()]
+    table_counts = load_table_counts(pipeline, *table_names)
+    assert set(table_counts.keys()) > set(tables)
+    assert table_counts["employees"] >= 31
+    assert table_counts["absences"] >= 6
```

Comment:
add tests for incremental loading [as in chess source](https://github.com/dlt-hub/verified-sources/blob/master/tests/chess/test_chess_source.py#L53): it checks loaded packages 
`assert_load_info(info, expected_load_packages=0)` 
and expects 0

Reply:
There is an issue with Personio API, when filtering by the `last_modified_at` field, I need to use the `updated_since` parameters 2s after the latest value, e.g.

The greatest value for `last_modified_at` was 2023-11-20T17:36:08:

- updated_since: 2023-11-20T17:36:08 brings one record;
- updated_since: 2023-11-20T17:36:09 brings one record;
- updated_since: 2023-11-20T17:36:10 empty data.

I will left it documented in the tests but let me know if you think that I should change something in the incremental logic.

Reply:
i think it's fine. dlt will merge them by id anyway, these are only one or two samples, so I think this will not affect the efficiency